core/backend/src/ViewStore.ts

```diff
@@ -0,0 +1,1397 @@
+ *
+ * Views may optionally have a thumbnail, paired via the View's Id. Thumbnails are stored in the "thumbnails" table.
+ *
+ * Note: All ElementIds and ModelIds in ModelSelectors, CategorySelectors, DisplayStyles, Timelines, etc. are converted to guid-based identifiers when stored in the ViewStore.
+ * They are then remapped back to their Ids when loaded from the ViewStore. This allows the ViewStore to be used with more than one iModel,
+ * provided that the same Guids are used in each iModel. This is done by storing the set of unique Guids in the "guids" table, and then
+ * creating a reference to the row in the "guids" table via the special Id prefix "^". For example, if a category selector contains the
+ * Id "0x123", then the guid from element 0x123 is stored in the "guids" table, and the category selector is stored with the rowId of the entry
+ * in the guid table (e.g. "^1w"). When the category selector is loaded from the ViewStore, the guid is looked up in the "guids" table and
+ * the iModel is queried for the element with that guid. That element's Id (which may or may not be 0x123) is then returned in the category selector.
+ *
+ * @beta
+ */
+export namespace ViewStore {
+
+  export const tableName = {
+    categorySelectors: "categorySelectors",
+    displayStyles: "displayStyles",
+    viewGroups: "viewGroups",
+    guids: "guids",
+    modelSelectors: "modelSelectors",
+    taggedViews: "taggedViews",
+    tags: "tags",
+    thumbnails: "thumbnails",
+    timelines: "timelines",
+    searches: "searches",
+    views: "views",
+  } as const;
+
+  /** data for a Thumbnail */
+  type ThumbnailData = Uint8Array;
+
+  /** A row in a table. 0 means "not present" */
+  export type RowId = number;
+
+  /** a string representation of a row in a table of a ViewStore. Will be a base-36 integer with a leading "@" (e.g."@4e3") */
+  type RowString = string;
+
+  /** a string representation of a row in the Guid table. Will be a base-36 integer with a leading "^" (e.g."^4e3") */
+  type GuidRowString = string;
+
+  /** common properties for all tables */
+  interface TableRow {
+    name?: string;
+    json: string;
+    owner?: string;
+  }
+  type DisplayStyleRow = TableRow;
+  type SelectorRow = TableRow;
+  type TagRow = TableRow;
+  type SearchRow = TableRow;
+  type TimelineRow = TableRow;
+
+  /** a row in the "views" table */
+  interface ViewRow extends MarkRequired<TableRow, "name"> {
+    className: string;
+    groupId: RowId;
+    isPrivate?: boolean;
+    modelSel?: RowId;
+    categorySel: RowId;
+    displayStyle: RowId;
+  }
+
+  /** a row in the "viewGroups" table  */
+  interface ViewGroupRow extends MarkRequired<TableRow, "name"> {
+    parentId: RowId;
+    defaultViewId?: RowId;
+  }
+
+  /** a row in the "thumbnails" table */
+  interface ThumbnailRow {
+    viewId: RowId;
+    data: ThumbnailData;
+    format: ThumbnailFormatProps;
+    owner?: string;
```

Comment:
What's the purpose of allowing thumbnails to have a different owner than the owner of the saved view that's related to the thumbnail?

Reply:
only that they're added/updated separately. I'm not even sure that `owner` will be used for anything other than views.

Reply:
Is there a reason why all tables have to contain an owner column then?

Reply:
Not really. I just figured it could be used to see who last changed it. Probably shouldn't be called "owner" though. You are very welcome to change that if you want!

Reply:
I do want to talk to you and @wgoehrig about how to get the "identity" (some unique string that isn't email address) of the person performing an action in the frontend. What do you to today?

Reply:
So the token has a subject that we can use as a user id. That's what we do today for saved views api.

Reply:
Yes, we can use the user's GUID for the owner field in the database. Is that what you use today? But, when we want to show the identity to a person (e.g. "who owns this view?"), we need another string that means something to them. It should be a user-chosen string (like your github id) and should **not** be their email address. I'm guessing lots of things can use it with an "@" prefix, like we do here.